srcpkgs/sparse/template

```diff
@@ -0,0 +1,30 @@
+# Template file for 'sparse'
+pkgname=sparse
+version=0.5.2
+revision=1
+build_style=gnu-makefile
+make_use_env=yes
+short_desc="Semantic Parser for C"
+maintainer="Anthony Iliopoulos <ailiop@altatus.com>"
+license="MIT"
+homepage="https://git.kernel.org/pub/scm/devel/sparse/sparse.git"
+distfiles="${homepage}/snapshot/${pkgname}-${version}.tar.gz"
+checksum=5ca5b7be7240a97fbdc6194f18d64837f9af070aeb0c3052f0a5d014acf1d048
+
+pre_build() {
+	make_build_args="CC=$CC LD=$CC AR=$AR"
```

Comment:
`make_build_args` doesn't need to go inside `pre_build()` (it usually goes below `build_style`)

Reply:
But when you put `make_build_args` in the main template body `$CC`, etc. haven’t been set, yet.

Reply:
They should be, based on $XBPS_CROSS_BUILD

Reply:
Indeed the toolchain variables are set by `setup_pkg()` after the template is sourced, so for the particular usage of `make_build_args` it needs to go into the `pre_build()` step. 

Alternatively it could be set within `do_build()`, but since there is no need to override any other invocation params of `make` it looks cleaner inside `pre_build()`.

Reply:
No, this case is special, because it uses the `$CC` and `$AR` variables. xbps-src sets them after `make_build_args` is read in if it was in the main template body. So these variables would be empty.

Reply:
Oh, I see, sorry for all the noise.